rest-api/test/unit_test/api_test/participant_summary_api_test.py

```diff
@@ -217,6 +219,59 @@ def post_demographics_questionnaire(self,
       url = 'Participant/%s/QuestionnaireResponse' % participant_id
       return self.send_post(url, request_data=response_data)
 
+  def test_csv_output(self):
+
+    SqlTestBase.setup_codes([PMI_SKIP_CODE], code_type=CodeType.ANSWER)
+    questionnaire_id = self.create_demographics_questionnaire()
+    t1 = TIME_1
+
+    def setup_participant(when, providerLink=self.provider_link):
+      # Set up participant, questionnaire, and consent
+      with FakeClock(when):
+        participant = self.send_post('Participant', {"providerLink": [providerLink]})
+        participant_id = participant['participantId']
+        self.send_consent(participant_id)
+        # Populate some answers to the questionnaire
+        answers = {
+          'race': RACE_WHITE_CODE,
+          'genderIdentity': PMI_SKIP_CODE,
+          'firstName': self.fake.first_name(),
+          'middleName': self.fake.first_name(),
+          'lastName': self.fake.last_name(),
+          'zipCode': '78751',
+          'state': PMI_SKIP_CODE,
+          'streetAddress': '1234 Main Street',
+          'city': 'Austin',
+          'sex': PMI_SKIP_CODE,
+          'sexualOrientation': PMI_SKIP_CODE,
+          'phoneNumber': '512-555-5555',
+          'recontactMethod': PMI_SKIP_CODE,
+          'language': PMI_SKIP_CODE,
+          'education': PMI_SKIP_CODE,
+          'income': PMI_SKIP_CODE,
+          'dateOfBirth': datetime.date(1978, 10, 9),
+          'CABoRSignature': 'signature.pdf',
+        }
+      self.post_demographics_questionnaire(participant_id, questionnaire_id, time=when, **answers)
+      return participant
+
+    [setup_participant(t1) for _ in range(5)]
+    response = self.send_get_csv('ParticipantSummary?_output=csv')
+    reader = StringIO.StringIO(response)
+    row = UnicodeDictReader(reader)
+    row = row.next()
+    headers = [k for k, _ in row.iteritems()]
+    for i in headers:
+      assert i in ps_full_data_headers
+    self.assertEqual(row['City'], 'Austin')
```

Comment:
You should be able to just make a big expectedRow dict with exactly what you expect in the response, and assert it matches row here... or if you prefer, read in CSV from a file. Either way, it will need to check all of the columns.

Reply:
done